src/test/integ/messages.test.ts

```diff
@@ -0,0 +1,123 @@
+    await wait();
+}
+
+describe("Message output", () => {
+    let client: NeovimClient;
+
+    before(async () => {
+        client = await attachTestNvimClient();
+        // Just for testing consistency with pre-0.10 versions:
+        await client.setOption("cmdheight", 0);
+        await openTextDocument({ content: "" });
+    });
+
+    after(async () => {
+        await closeNvimClient(client);
+        await closeAllActiveEditors();
+    });
+
+    afterEach(async () => {
+        await client.command("messages clear");
+        await hideOutputPanel();
+    });
+
+    it("should reveal output panel with contents", async () => {
+        await sendCommandLine("echo 1 | echom 2 | echo 3 | echom 4");
+        await wait();
+        let outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "1\n2\n3\n4\n");
+        await hideOutputPanel();
+
+        await sendCommandLine("messages");
+        await wait();
+        outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "echomsg: 2\nechomsg: 4\n");
+        await hideOutputPanel();
+
+        await sendCommandLine("echo 5 | echo 6 | echo 7");
+        await wait();
+        outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "5\n6\n7\n");
+    });
+
+    it("should reveal after first line", async () => {
+        await sendCommandLine("echom 1 | sleep 1 | echom 2 | echom 3");
+
+        // only one line written at first, should not be revealed yet
+        let outputEditor = findOutputChannel();
+        assert.equal(outputEditor, undefined);
+
+        await wait(1400);
+        outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "1\n2\n3\n");
+        await hideOutputPanel();
+
+        await sendCommandLine("messages");
+        await wait();
+        outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "echomsg: 1\nechomsg: 2\nechomsg: 3\n");
+    });
+
+    it("should clear history", async () => {
+        let outputEditor = findOutputChannel();
+        await sendCommandLine("echom 1 | echom 2 | echom 3");
+        await wait();
+        await sendCommandLine("messages");
+        await wait();
+        outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "echomsg: 1\nechomsg: 2\nechomsg: 3\n");
+
+        await sendCommandLine("messages clear");
+        await wait();
+        await sendCommandLine("messages");
+        await wait();
+        outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "");
+    });
+
+    it("should reveal for return prompt", async () => {
+        await sendNeovimKeys(client, "/foobar\n");
+        await wait();
+        let outputEditor = findOutputChannel();
+        assert.equal(outputEditor?.document.getText(), "/foobar             \nE486: Pattern not found: foobar\n");
```

Comment:
I totally get why you made this change (I did it, too), but this is likely going to frustrate a lot of users. https://github.com/vscode-neovim/vscode-neovim/issues/2046

Unfortunately, it seems sane to do what you're doing (`return_prompt` implies we should have something meaningful displayed to the user), but when `ext_messages` is enabled, nvim sends us lots of `return_prompt`s that one wouldn't otherwise expect (e.g, after a missing pattern). This is ultimately why I just ignored `return_prompt` in the end :/

Reply:
Oops, you're right, in my arrogance I thought these changes would work around that issue but I forgot to actually compare `/`-search behavior with nvim itself and also prioor version of the extension. A couple cases I should probably test for both 0.9 and 0.10:

- [ ] search with default nvim configuration
- [ ] search with misc. `cmdheight` settings
- [ ] maybe try different `shortmess` settings? 
- [ ] Are there other easy ways of triggering the hit-enter prompt??

In #2029 you suggested possibly adding a configuration setting for hit-enter prompt behavior... given this keeps coming up, do you think it makes sense to consider adding this now? For this PR I will try to maintain the behavior as before (to prevent another #2046), but maybe a follow-up could include allowing users to opt-in to hit-enter prompts revealing the output panel, plus maybe some docs about `shortmess` would help?

Reply:
> In https://github.com/vscode-neovim/vscode-neovim/pull/2029 you suggested possibly adding a configuration setting for hit-enter prompt behavior... given this keeps coming up, do you think it makes sense to consider adding this now?

I'm not sure, but I'm leaning towards no. If nvim's return_prompt behavior was consistent between clients, maybe (which was my understanding when I wrote that), but I suspect most users don't want an output panel for literally every output.

Reply:
Fair enough... the good news is it looks like upstream they are planning to overhaul this and reduce the frequency significantly: https://github.com/neovim/neovim/issues/22478

I'm not sure what the exact implications for external `ui-messages` consumers will be, but maybe it will help with this problem in general.

Reply:
hoo boy, okay. It turns out even after removing the `return_prompt` handling, this PR will still reveal on failed searches for anyone whose `cmdheight` is < 2, because there's no `msg_clear` between the `/<pattern>` message and the following error message, and `replace_last` is also set to false 😭 

```
msg_clear [ [] ]
msg_show [ [ '', [ [ 0, '/asdkfjhqwe             ' ] ], false ] ]
Flushing message buffer: [ '/asdkfjhqwe             ' ]
msg_show [
  [ 'emsg', [ [ 90, 'E486: Pattern not found: asdkfjhqwe' ] ], false ],
  [
    'return_prompt',
    [ [ 166, 'Press ENTER or type command to continue' ] ],
    false
  ]
]
Flushing message buffer: [ '/asdkfjhqwe             ', 'E486: Pattern not found: asdkfjhqwe' ]
```

I guess it's possible to set the default `cmdheight` back to 2, kinda like the behavior before 4ca8fbf70edfaaf080823d583e8d2105ace0c06e ? It might still be a breaking change for people who've got `cmdheight` explicitly set in their configuration in that case... 

The only other option I can think of is trying to display a hint the first time to let people know they can increase cmdheight to suppress the output panel, or something like that, to help avoid the influx of unhappy users filing issues... I'll keep thinking about different possible options I guess

Reply:
I've been thinking about this, and I think that if we're going to tell users to customize cmdheight, is there much difference between it and us making a configuration value? I'm not sure which I prefer. I think `cmdheight` makes the most sense, because then the solution is fully in nvim's control.

Reply:
Ok, I tried to implement this configuration hint popup and it mostly works, but I realized while testing against 0.9.5 that since `cmdheight` isn't configurable for the attached UI until 0.10, we always end up with `cmdheight=0` and there's no way to suppress the output reveal. I suppose that might be an argument in favor of a `vscode-neovim` configuration? 

Alternatively, maybe just suggest they upgrade to 0.10+ to make `cmdheight` work as expected is simpler, idk.

Reply:
Maybe a good compromise is to make an option that accepts a minimum number of lines before output is revealed, with a default of "cmdheight", (or null if necessary), which will fall back to the nvim value

Reply:
Also, I hope it goes without saying, but I appreciate the care you've put into this!!